src/librbd/crypto/CryptoObjectDispatch.cc

```diff
@@ -151,13 +155,14 @@ bool CryptoObjectDispatch<I>::write(
                  << object_off << "~" << data.length() << dendl;
   ceph_assert(m_crypto != nullptr);
 
-  m_crypto->encrypt(
-          std::move(data),
+  bufferlist data_copy = data;
```

Comment:
Nit: was there some reason why you need to copy it?

Reply:
I have an rvalue reference (data), but now CryptoInterface needs a pointer.
As I understand, you cannot take the address of an rvalue, and thus I need some temporary variable whose address I can take.
Can you write down the alternative you thought about?

Reply:
You should definitely be allowed to take a pointer to this rvalue. Are you seeing a compile error for some reason?

Reply:
Well I actually googled "taking address of rvalue" before trying to compile, which lead me to the thought that it is problematic.
Anyway, it seems to compile. Pushed the change now.

Reply:
Thanks -- it's technically an rvalue reference which is why it should be legal (vs trying to take the address of an rvalue returned from a function).